hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/PathConvertingFileSystem.java

```diff
@@ -154,19 +154,8 @@ private Path from(Path path) {
         return fromFunc.apply(path);
     }
 
-    private FileStatus toReturnFileStatus(FileStatus status) throws IOException {
-        // same as FileStatus copy constructor
-        return new FileStatus(
-                status.getLen(),
-                status.isDirectory(),
-                status.getReplication(),
-                status.getBlockSize(),
-                status.getModificationTime(),
-                status.getAccessTime(),
-                status.getPermission(),
-                status.getOwner(),
-                status.getGroup(),
-                status.isSymlink() ? status.getSymlink() : null, // getSymlink throws if file is not a symlink
-                from(status.getPath()));
+    private FileStatus toReturnFileStatus(FileStatus status) {
+        status.setPath(from(status.getPath()));
+        return status;
     }
```

Comment:
Is there a test case that covers the behavior we want here?

Reply:
Yeah, PathConvertingFileSystemTest covers the conversion here: https://github.com/palantir/hadoop-crypto/blob/d0853ab96feca47a7e6f385f52bc71d8a2a54357/hadoop-crypto/src/test/java/com/palantir/crypto2/hadoop/PathConvertingFileSystemTest.java#L121-L135

Reply:
Also testing e.g. the object reference felt too much like an implementation detail. But can add.

Reply:
On further thought testing the new behavior makes sense: added in 410ff78.